### ISSUE TITLE
Obspack in Flexgrid model (v12.7.0) [BUG/ISSUE]

### ISSUE THREAD

**Issue by lshen2009** [geoschem/geos-chem#337]
When I turn on the obspack in CH4 simulations, nothing is archived. I am using the Flexigrid (12.7.0) model.  I think the problem is caused by a bug in the input_mod.F related to reading the obspack in the flexgrid model.
```
%%% OBSPACK MENU %%%    :
Turn on ObsPack diag?   : T
Quiet logfile output    : T
```
But when I go to main.F, and add this
```
print *, 'lshen_Input_Opt%Do_ObsPack',Input_Opt%Do_ObsPack
```
it prints  'F'.

Here is my run directory in Cannon   /n/holyscratch01/jacob_lab/lshen/CH4/GEOS-Chem/Flexgrid/Global_4x5_obspack
My code  ./CodeDir

Thanks.

**Comment by yantosca** [comment]
Thanks for writing.  Can you first try recompiling & rerunning with the debug flags on to see if there isn't an out-of-bounds error that is tripping the flag?  Use DEBUG=y BOUNDS=y FPEX=y.

**Comment by lshen2009** [comment]
I just tried and there isn't an out-of-bounds error.  But this bug still exists. I printed it in input_mod.F.
Input_Opt%Do_ObsPack F

**Comment by yantosca** [comment]
Which log file shows the error in your run directory?

**Comment by lshen2009** [comment]
I submit the job run.sh
The output is in log.merra2

Thanks.

**Comment by yantosca** [comment]
Thanks for writing.  I think the call to READ_OBSPACK_MENU was somehow removed from routine READ_INPUT_FILE, perhaps in a Git merge.  That seems to be why we are not reading the OBSPACK menu.

@msulprizio is more familiar with the CH4 code than I am but I wonder if this is also in the standard code.  We can push a fix in 12.9.0.

**Comment by yantosca** [comment]
Thanks for writing.  I think the call to READ_OBSPACK_MENU was somehow removed from routine READ_INPUT_FILE, perhaps in a Git merge.  That seems to be why we are not reading the OBSPACK menu.

@msulprizio is more familiar with the CH4 code than I am but I wonder if this is also in the standard code.  We can push a fix in 12.9.0.

**Comment by lshen2009** [comment]
Thanks, Bob. Should I contact Melissa about how to fix this bug? Thanks.

**Comment by msulprizio** [comment]
@lshen2009 I can add a fix to our methane workflow code

**Comment by yantosca** [comment]
I'll add a fix for this into 12.9.0.  We just have to add the call to READ_OBSPACK_MENU from routine READ_INPUT_FILE in input_mod.F (or *.F90, depending on the version).  This seems to have been removed inadvertently in a Git merge.  My apologies.

**Comment by msulprizio** [comment]
This is now resolved in commit ca1407108a85fe0c769d26ad9d638d8c04e38345 and will be included in 12.9.0.
